src/app/components/TradeProfit/index.tsx

```diff
@@ -1,70 +1,114 @@
 import { bignumber } from 'mathjs';
-import { Tooltip } from '@blueprintjs/core';
+import { Tooltip2 } from '@blueprintjs/popover2';
 import { useTranslation } from 'react-i18next';
 import { TradingPosition } from '../../../types/trading-position';
-import {
-  toNumberFormat,
-  weiToNumberFormat,
-} from '../../../utils/display-text/format';
+import { backendUrl, currentChainId } from 'utils/classifiers';
+import { toWei } from 'utils/blockchain/math-helpers';
+import { TradingPair } from 'utils/models/trading-pair';
+import { useAccount } from '../../hooks/useAccount';
+import { toNumberFormat, weiToNumberFormat } from 'utils/display-text/format';
 import { Asset } from '../../../types/asset';
 import { translations } from '../../../locales/i18n';
 
 interface Props {
+  asset: Asset;
   entryPrice: string;
   closePrice: string;
-  profit: string;
   position: TradingPosition;
-  asset: Asset;
+  positionSize: string;
+  loanToken: string;
+  pair: TradingPair;
+  loanId: string;
 }
 
 export function TradeProfit(props: Props) {
-  let change = bignumber(bignumber(props.closePrice).minus(props.entryPrice))
-    .div(props.entryPrice)
-    .mul(100)
-    .toNumber();
-  if (props.position === TradingPosition.SHORT) {
-    change = bignumber(bignumber(props.entryPrice).minus(props.closePrice))
-      .div(props.entryPrice)
-      .mul(100)
-      .toNumber();
-  }
+  const account = useAccount();
+  const [profit, setProfit] = useState<string>('');
+  const [profitDirection, setProfitDirection] = useState<number>(0);
+  const prettyPrice = amount => {
+    return props.loanToken !== props.pair.shortAsset
+      ? toWei(
+          bignumber(1)
+            .div(amount)
+            .mul(10 ** 8),
+        )
+      : toWei(bignumber(amount).div(10 ** 8));
+  };
+
+  fetch(backendUrl[currentChainId] + '/events/trade/' + account)
+    .then(response => {
+      return response.json();
+    })
+    .then(loanEvents => {
+      loanEvents.events.forEach(events => {
+        if (events.loanId === props.loanId) {
+          const entryPrice = prettyPrice(events.data[0].collateralToLoanRate);
+          const closePrice = prettyPrice(
+            events.data.slice(-1).pop().collateralToLoanRate,
+          ); // getting the last element in data
+          const positionSize = events.data[0].positionSize;
+
+          //LONG position
+          let change = bignumber(bignumber(closePrice).minus(entryPrice))
+            .div(entryPrice)
+            .mul(100)
+            .toNumber();
+
+          //SHORT position
+          if (props.position === TradingPosition.SHORT) {
+            change = bignumber(bignumber(entryPrice).minus(closePrice))
+              .div(entryPrice)
+              .mul(100)
+              .toNumber();
+          }
+          setProfit(
+            bignumber(change).mul(bignumber(positionSize)).div(100).toFixed(0),
+          );
+          setProfitDirection(change);
+        }
+      });
+    })
+    .catch(console.error);
```

Comment:
Put it to `useEffect` to make sure it's being called only once, cancel request when component unmounts.